src/components/Spinner.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { View, Animated, Easing } from 'react-native';
+import styled from '@stream-io/styled-components';
+import { themed } from '../styles/theme';
+
+const AnimatedView = Animated.createAnimatedComponent(View);
+
+const Circle = styled(AnimatedView)`
+  height: 30px;
+  width: 30px;
+  margin: 5px;
+  border-width: 2px;
+  border-radius: 30px;
+  justify-content: center;
+  border-color: ${({ theme }) => theme.colors.primary};
+  border-right-color: transparent;
+  ${({ theme }) => theme.spinner.css}
+`;
+
+export const Spinner = themed(
+  class Spinner extends React.PureComponent {
+    static themePath = 'spinner';
+    state = {
+      rotateValue: new Animated.Value(0),
+    };
+
+    componentDidMount() {
+      this._start();
+    }
+
+    _start = () => {
+      Animated.loop(
+        Animated.timing(this.state.rotateValue, {
+          toValue: 1,
+          duration: 800,
+          easing: Easing.linear,
```

Comment:
Ohhh lets use `useNativeDriver: true`. Animations run quite smoothly as per their doc since frames don't need to go over the bridge on every frame !! - http://facebook.github.io/react-native/blog/2017/02/14/using-native-driver-for-animated